.github/workflows/main.yml

```
@@ -217,13 +179,75 @@ jobs:
         run: make typecheck-contrib
       - name: Run the TypeScript typechecker for scripts
         run: make typecheck-scripts
-
+      - name: Run the GitHub Actions linter
+        run: make lint-actions
       # This step runs at the end, since it is common for it to fail in
       # dependabot PRs, but we still want all other tests above to run
       # in those cases.
       - name: Check for duplicate Node dependencies
         run: yarn dedupe --check
 
+  native-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install OS packages
+        uses: awalsh128/cache-apt-pkgs-action@v1.5.0
+        with:
+          packages: graphviz libgraphviz-dev
+          version: 1.0
+          execute_install_scripts: true
+
+      - name: Install additional dependencies
+        run: |-
+          sudo curl -fsSL https://github.com/hadolint/hadolint/releases/download/v2.12.0/hadolint-Linux-x86_64 -o /bin/hadolint && \
+          sudo chmod +x /bin/hadolint && sudo curl -fsSL https://d2lang.com/install.sh | sh -s -- && \
+          sudo apt-get install -y shellcheck
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+        with:
+          enable-cache: true
+          cache-dependency-glob: 'images/plbase/python-requirements.txt'
+      - name: Install Python dependencies
+        env:
+          UV_SYSTEM_PYTHON: 1
+        run: uv pip install -r images/plbase/python-requirements.txt
+
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+      - name: Set up Yarn
+        run: yarn workspaces focus prairielearn
+        # We only install the dependencies in the root workspace
```

Nits (consistent naming; comment above the line it's about; comment explaining why and not just what):

```suggestion
      - name: Install Node dependencies
        # We only install the dependencies in the root workspace to avoid
        # installing dependencies that won't be used.
        run: yarn workspaces focus prairielearn
```

---

Ah shoot, I had merge on. My fault, I shouldn't have enabled that.